From 8b1f651e2dc56416d96e1d8b49eae9649e965038 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Cerveau?= <scerveau@igalia.com>
Date: Tue, 30 May 2023 09:38:57 +0200
Subject: [PATCH 154/232] va: use GstH264Level enum in _va_h264_level_limits

The 1B value has been changed to 9 instead of 11 in the list
of level limits.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4737>
---
 sys/va/gstvah264enc.c | 40 ++++++++++++++++++++--------------------
 1 file changed, 20 insertions(+), 20 deletions(-)

diff --git a/sys/va/gstvah264enc.c b/sys/va/gstvah264enc.c
index 586540039..7bd54e62b 100644
--- a/sys/va/gstvah264enc.c
+++ b/sys/va/gstvah264enc.c
@@ -310,26 +310,26 @@ struct _GstVaH264LevelLimits
 /* *INDENT-OFF* */
 static const GstVaH264LevelLimits _va_h264_level_limits[] = {
   /* level   idc   MaxMBPS   MaxFS   MaxDpbMbs  MaxBR   MaxCPB  MinCr */
-  {  "1",    10,   1485,     99,     396,       64,     175,    2 },
-  {  "1b",   11,   1485,     99,     396,       128,    350,    2 },
-  {  "1.1",  11,   3000,     396,    900,       192,    500,    2 },
-  {  "1.2",  12,   6000,     396,    2376,      384,    1000,   2 },
-  {  "1.3",  13,   11880,    396,    2376,      768,    2000,   2 },
-  {  "2",    20,   11880,    396,    2376,      2000,   2000,   2 },
-  {  "2.1",  21,   19800,    792,    4752,      4000,   4000,   2 },
-  {  "2.2",  22,   20250,    1620,   8100,      4000,   4000,   2 },
-  {  "3",    30,   40500,    1620,   8100,      10000,  10000,  2 },
-  {  "3.1",  31,   108000,   3600,   18000,     14000,  14000,  4 },
-  {  "3.2",  32,   216000,   5120,   20480,     20000,  20000,  4 },
-  {  "4",    40,   245760,   8192,   32768,     20000,  25000,  4 },
-  {  "4.1",  41,   245760,   8192,   32768,     50000,  62500,  2 },
-  {  "4.2",  42,   522240,   8704,   34816,     50000,  62500,  2 },
-  {  "5",    50,   589824,   22080,  110400,    135000, 135000, 2 },
-  {  "5.1",  51,   983040,   36864,  184320,    240000, 240000, 2 },
-  {  "5.2",  52,   2073600,  36864,  184320,    240000, 240000, 2 },
-  {  "6",    60,   4177920,  139264, 696320,    240000, 240000, 2 },
-  {  "6.1",  61,   8355840,  139264, 696320,    480000, 480000, 2 },
-  {  "6.2",  62,  16711680,  139264, 696320,    800000, 800000, 2 },
+  {  "1",    GST_H264_LEVEL_L1,   1485,     99,     396,       64,     175,    2 },
+  {  "1b",   GST_H264_LEVEL_L1B,   1485,     99,     396,       128,    350,    2 },
+  {  "1.1",  GST_H264_LEVEL_L1_1,   3000,     396,    900,       192,    500,    2 },
+  {  "1.2",  GST_H264_LEVEL_L1_2,   6000,     396,    2376,      384,    1000,   2 },
+  {  "1.3",  GST_H264_LEVEL_L1_3,   11880,    396,    2376,      768,    2000,   2 },
+  {  "2",    GST_H264_LEVEL_L2,   11880,    396,    2376,      2000,   2000,   2 },
+  {  "2.1",  GST_H264_LEVEL_L2_1,   19800,    792,    4752,      4000,   4000,   2 },
+  {  "2.2",  GST_H264_LEVEL_L2_2,   20250,    1620,   8100,      4000,   4000,   2 },
+  {  "3",    GST_H264_LEVEL_L3,   40500,    1620,   8100,      10000,  10000,  2 },
+  {  "3.1",  GST_H264_LEVEL_L3_1,   108000,   3600,   18000,     14000,  14000,  4 },
+  {  "3.2",  GST_H264_LEVEL_L3_2,   216000,   5120,   20480,     20000,  20000,  4 },
+  {  "4",    GST_H264_LEVEL_L4,   245760,   8192,   32768,     20000,  25000,  4 },
+  {  "4.1",  GST_H264_LEVEL_L4_1,   245760,   8192,   32768,     50000,  62500,  2 },
+  {  "4.2",  GST_H264_LEVEL_L4_2,   522240,   8704,   34816,     50000,  62500,  2 },
+  {  "5",    GST_H264_LEVEL_L5,   589824,   22080,  110400,    135000, 135000, 2 },
+  {  "5.1",  GST_H264_LEVEL_L5_1,   983040,   36864,  184320,    240000, 240000, 2 },
+  {  "5.2",  GST_H264_LEVEL_L5_2,   2073600,  36864,  184320,    240000, 240000, 2 },
+  {  "6",    GST_H264_LEVEL_L6,   4177920,  139264, 696320,    240000, 240000, 2 },
+  {  "6.1",  GST_H264_LEVEL_L6_1,   8355840,  139264, 696320,    480000, 480000, 2 },
+  {  "6.2",  GST_H264_LEVEL_L6_2,  16711680,  139264, 696320,    800000, 800000, 2 },
 };
 /* *INDENT-ON* */
 
-- 
2.40.1

